### PR TITLE
Porting AYTF fix done in PR #2376 to JS, this is missed in that PR.

### PR DESCRIPTION
**Pull request by penmetsaa**

### CODE REVIEW

File: javascript/i18n/phonenumbers/asyoutypeformatter.js [hunk 1]
```diff
@@ -726,7 +726,21 @@ i18n.phonenumbers.AsYouTypeFormatter.prototype.attemptToFormatAccruedDigits_ =
...
+      // in that way.
+      var fullOutput = this.appendNationalNumber_(formattedNumber);
+      var formattedNumberDigitsOnly =
+	  i18n.phonenumbers.PhoneNumberUtil.normalizeDiallableCharsOnly(fullOutput);
+      if (formattedNumberDigitsOnly == this.accruedInputWithoutFormatting_) {
+          // If it's the same (i.e entered number and format is same), then it's
+          // safe to return this in formatted number as nothing is lost / added.
+          return fullOutput;
+      }
+      return fullOutput;
```
Comment (lararennie) [inline javascript/i18n/phonenumbers/asyoutypeformatter.js]: why this return fullOutput here? we don't have this in java

**Comment by lararennie** [comment]
You will also need to re-compile the demo.

**Comment by lararennie** [comment]
Running demo without fix:
https://screenshot.googleplex.com/i7TZBcn43Nj

File: javascript/i18n/phonenumbers/asyoutypeformatter.js [hunk 2]
```diff
@@ -726,7 +726,20 @@ i18n.phonenumbers.AsYouTypeFormatter.prototype.attemptToFormatAccruedDigits_ =
...
-      return this.appendNationalNumber_(formattedNumber);
+      // Check that we didn't remove nor add any extra digits when we matched
+      // this formatting pattern. This usually happens after we entered the last
+      // digit during AYTF. Eg: In case of MX, we swallow mobile token (1) when
+      // formatted but AYTF should retain all the number entered and not change
+      // in order to match a format (of same leading digits and length) display
+      // in that way.
+      var fullOutput = this.appendNationalNumber_(formattedNumber);
+      var formattedNumberDigitsOnly =
+	  i18n.phonenumbers.PhoneNumberUtil.normalizeDiallableCharsOnly(fullOutput);
```
Comment (lararennie) [inline javascript/i18n/phonenumbers/asyoutypeformatter.js]: remove the tab on this line and replace with spaces

**Comment by lararennie** [comment]
> You will also need to re-compile the demo.

See instructions here if you need them:
https://github.com/google/libphonenumber/blob/master/javascript/README.md

**Comment by lararennie** [comment]
> Running demo without fix:
> https://screenshot.googleplex.com/i7TZBcn43Nj

With fix:
https://screenshot.googleplex.com/0Yqkc2q0iey

**Comment by lararennie** [comment]
Right: you need to fix the tab and compile the demo (to check the javascript is okay) and add the compiled demo to this PR then this LGTM.

**Comment by penmetsaa** [comment]
Done Lara. Thanks so much :)

**Pull request merged**